File: src/netconsole/models/api/features.py
from __future__ import annotations

from pydantic import Field

from netconsole.models.api.common import ApiModel


class RendererFeatureStateDTO(ApiModel):
    feature_id: str
    visible: bool
    enabled: bool


class RendererFeatureStateListDTO(ApiModel):
    items: list[RendererFeatureStateDTO] = Field(default_factory=list)


__all__ = ["RendererFeatureStateDTO", "RendererFeatureStateListDTO"]

File: apps/desktop_renderer/src/api/features.ts
import { apiRequest } from './client'

export interface RendererFeatureState {
  feature_id: string
  visible: boolean
  enabled: boolean
}

export async function getRendererFeatureStates(): Promise<RendererFeatureState[]> {
  const response = await apiRequest<{ items: RendererFeatureState[] }>('/api/features')
  return response.items
}
